core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -3611,6 +3611,20 @@ void checkCorrelatedMapSubQuery(boolean expand) {
         .convertsTo("${planConverted}");
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-4683">[CALCITE-4683]
+   * In-list to join causes field datatypes not matched</a>. */
+  @Test void testInToSemiJoinWithNewProject() {
+    // Before this bug fixed, converting the sql to rel will fail.
```

Comment:
I'd rather drop this comment here and add the stack trace of the error in the description of the Jira ticket, it will help people understand if they are hitting the same bug as you.

Reply:
OK, I will do that.